### DISCUSSION TITLE
I need help understanding the entrie ebusd adapter shield v5 ecosystem

### DISCUSSION THREAD

**Discussion by MartinMajewski** [john30/ebusd#1279] [General]
Hi all,

I scanned the discussions here and read through the documentation, but I am still confused and don't understand how the shield v5 is intended to be used.

First, I have Home Assistant (HA) running on a Yellow Board far away from my Ochsner Heatpump that I want to read and control via ebus. I ordered the Shield v5, got it running as a stand-alone WiFi client, and can access its web interface and the REPL CLI. It is up to date with firmware build 20240518.

Okay, but now what?

I want to connect the shield as a stand-alone device (as advertised) to the heat pump's ebus port.
The communication with HA should be done via MQTT

Now, my questions:

1. How do I start the MQTT client on the Shield v5? I tried different things on the REPL command line, but the "mqtt" command looks quite limited.
2. How to configure the connection to my MQTT broker running on HA (URI, Port, User, and Password)
3. What HA integration to use?

Or do I misunderstand the entire setup, and the Shield V5 cannot be used without an underlying Raspberry Pi? But then, I find the information on the product's side misleading.

**Comment by Koky05** [comment]
At first you need [eBUS Deamon](https://github.com/john30/ebusd). There is several options how to run it:
 - as [native app](https://github.com/john30/ebusd?tab=readme-ov-file#installation) compiled directly on some server
 - as standalone [docker container](https://github.com/john30/ebusd?tab=readme-ov-file#docker-image) but then you need also another container with MQTT-broaker
 - as [HA Addon](https://github.com/LukasGrebe/ha-addons)
 
 All [configuration](https://github.com/john30/ebusd-configuration) for communication between eBUS adapter and eBUS Deamon is lead by deamon part.
 You do not need HA Integration because in every case you can use HA AutoDiscovery via MQTT. So only need is to connect HA to MQTT.
 But in your case it will be little complicated because for your heatpump I don't think exists configuration files to read data. So you need t create some for you.  [Closes config files](https://github.com/cybersmart-eu/ebusd-ochsner) based on discussion [#922](https://github.com/john30/ebusd/discussions/922)

**Comment by MartinMajewski** [comment]
Thanks for the response. I am confident that I can create a configuration for the pump after some research. As I have an OTE 3 terminal offering a web2con interface, I can sniff the traffic there and map with the tabular output.

What confused me is how the abused Shield V5 works. In the entire HA universe, every IoT device is handled differently, and HA has changed so much in the last two years that it's hard to keep up.

So, to verify that I understand it correctly (hopefully) now, please review the following statements:

1. HA uses the AddOn to connect to an ebus adapter like the ebusd Shield V5.
2. The AddOn uses the device string (e.g., ens:192.168.0.50:9999) to connect to the ebus device like the Shield V5.
3. The ebus device (Shield V5) does not need to be physically attached to the daemon's host device
4. The connection is made, in this case, via TCP over WiFi via the device string.
5. The AddOn reads the ebus data via TCP, packs them into an intermediate format like an MQTT message, and sends it, in this case, via localhost to the Mosquitto-Broker running on the same host system (HA), where it can be used directly as a heating source device.
6. I need to provide the device string, the MQTT broker information, etc., in some HA config files (but which ones and where? HA does not allow access to "/etc")

Am I, at least partially, correct?

Best wishes,
Martin

**Reply by MartinMajewski** [reply]
P.S.: The HA AddOn does not install the ebus daemon, does it? After installing it and connecting to HA via SSH, I cannot see any ebusd-related folders or files.

**Reply by Koky05** [reply]
HA Addon is based on eBUS Deamon docker version with some additional feature to allow it run as Addon under HA. So yes it install deamon. All HA Addons are docker containers. It did not creates any folder but it maps one from HA configuration folder. Most information how to work with it is in [documentation of Addon](https://github.com/LukasGrebe/ha-addons/blob/main/ebusd/DOCS.md#custom-csv-or-mqtt-cfg-files).
1. yes
2. yes
3. yes, it doesn't need to be physicali connected if your Shield V5 use WiFi or other way to connect to local network.
4. yes
5. yes
6. No. All information you should put into Addon configuration. Only config files should be placed into HA Config folder like is written in Addon documentation.

**Reply by MartinMajewski** [reply]
Thank you for your patience and reply.

Okay, I got it working. What tripped me was the statement in the documentation:

> Define either a USB or network device. Seperate configuration entries are available via the UI or can be configured manually using device: /dev/ttyAMA0 or network_device: enh:192.158.0.7:9999

There were no UI fields for the device nor mqtt settings on the configuration page of the add-on.
I accidentally clicked on the hamburger menu in the top right corner and switched to the YAML view, where I could enter all necessary information. Only then did the add-on start. There is certainly room for improvement in the documentation for HA noobs like myself! :-D

**Comment by GuyHarg** [comment]
Thanks you for the thread which was a big help to me tonight. I've got HA Addon version 23.2 installed and working, and my updater says there is 23.3 available but I can't seem to update it. Do I need to uninstall the Addon and reinstall it or has the add-on just not been updated for Xanthophyllum 23.3? Cheers, Guy
